Userland/Libraries/LibGUI/DynamicWidgetContainer.cpp

```diff
@@ -0,0 +1,464 @@
+            auto child_container = child_containers();
+
+            order_or_error.value().as_array().for_each([&](auto& section_label) {
+                for (auto& container : child_container) {
+                    if (container.section_label() == section_label.to_deprecated_string())
+                        new_child_order.append(container);
+                }
+            });
+
+            // Are there any children that are not known to our persisted order?
+            for (auto& container : child_container) {
+                // FIXME: Optimize performance and get rid of conains_slow so that this does not become a issue when a lot of child containers are used.
+                if (!new_child_order.contains_slow(container))
+                    new_child_order.append(container);
+            }
+
+            // Rearrange child widgets to the defined order.
+            auto childs = child_widgets();
+            for (auto& child : childs) {
+                if (new_child_order.contains_slow(child))
+                    child.remove_from_parent();
+            }
+
+            for (auto const& child : new_child_order)
+                add_child(*child);
+        } else {
+            int persisted_state = Config::read_i32(config_domain(), "DynamicWidgetContainers"sv, section_label(), (int)ViewState::Expanded);
+            set_view_state(static_cast<ViewState>(persisted_state));
+        }
+        update();
+    });
+}
+
+void DynamicWidgetContainer::set_section_label(String label)
+{
+    m_section_label = move(label);
+
+    auto label_widget = m_controls_widget->find_descendant_of_type_named<GUI::LabelWithEventDispatcher>("section_label");
+    label_widget->set_text(m_section_label);
+}
+
+void DynamicWidgetContainer::set_config_domain(String domain)
+{
+    m_config_domain = move(domain);
+    // FIXME: A much better solution would be to call the restore_view_state within a dedicated "initialization finished" method that is called by the gml runtime after that widget is ready.
+    //        We do not have such a method yet.
+    restore_view_state();
+}
+
+void DynamicWidgetContainer::set_detached_size(Gfx::IntSize const size)
+{
+    m_detached_size = { size };
+}
+
+void DynamicWidgetContainer::set_show_controls(bool value)
+{
+    m_show_controls = value;
+    m_controls_widget->set_visible(m_controls_widget->is_visible() && show_controls());
+    update();
+}
+
+void DynamicWidgetContainer::set_container_with_individual_order(bool value)
+{
+    m_is_container_with_individual_order = value;
+}
+
+void DynamicWidgetContainer::second_paint_event(PaintEvent&)
+{
+    GUI::Painter painter(*this);
+    painter.fill_rect_with_gradient(Gfx::IntRect(0, height() - 1, width(), height()), palette().threed_shadow1().lightened(), palette().button());
+
+    if (!m_is_dragging && !m_render_as_move_target)
+        return;
+
+    if (m_is_dragging) {
+        // FIXME: Would be nice if we could paint outside our own boundaries.
```

Comment:
The reason you can't is that the `painter(*this)` constructor above is clipping to this widget's bounds. I think `painter.clear_clip_rect()` would solve this but I haven't tested it myself. Might take some fiddling with.